client/src/services/userService.ts

```diff
@@ -22,34 +22,63 @@ export default class UserService {
     static currentUser: User | undefined;
 
     static async lockFront(password: string): Promise<string> {
-        return http.post('/user/lock', { password });
+        return http.post<string, {password:string}>('/user/lock', { password })
+            .then((b) => b)
+            .catch(() => {
+                toastService.toast(tr('user.errorWhileLocking'), 'error');
+                return '';
+            });
     }
 
     static async isLocked(): Promise<boolean> {
-        return http.get('/user/isLocked');
+        return http.get<boolean>('/user/isLocked')
+            .then((b) => b)
+            .catch(() => {
+                toastService.toast(tr('user.errorWhileLocking'), 'error');
```

Comment:
what does isLocked do?

Reply:
It locks the front end

Reply:
It checks if the application is locked (verrouiller) or not